tools/update-conda.py

```diff
@@ -0,0 +1,154 @@
+        return
+
+    dependencies = get_dependencies(pyproject_toml)
+
+    for platform_key, platform_value in platforms.items():
+        for python in pythons:
+            for tag in tags:
+                # Pin Python version
+                pinned_dependencies = {
+                    f"python={python}" if dep == "python" else dep
+                    for dep in dependencies
+                }
+                if tag == "-dev":
+                    dev_dependencies = get_dev_dependencies(pyproject_toml)
+                    print(f"Adding dev dependencies: {dev_dependencies}")
+                    pinned_dependencies = pinned_dependencies.union(dev_dependencies)
+                pinned_dependencies = sorted(pinned_dependencies)
+
+                for src in sources:
+                    env_file = source_dir / src / f"environment{tag}-{python}.yml"
+                    write_env_file(env_file, pinned_dependencies)
+                    lock_file = (
+                        source_dir / src / f"environment{tag}-{python}-{platform_value}"
+                    )
+                    print(
+                        f"Updating lock file for {env_file} at {lock_file}", flush=True
+                    )
+                    subprocess.run(
+                        [
+                            "conda-lock",
+                            "--channel",
+                            "conda-forge",
+                            "--kind",
+                            "env",
+                            "--platform",
+                            platform_key,
+                            "--file",
+                            str(env_file),
+                            "--lockfile",
+                            str(lock_file),
+                            "--filename-template",
+                            str(lock_file),
+                        ]
+                    )
+
+
+def get_dependencies(pyproject_toml: Path) -> list[str]:
+    graystull_config = Configuration("sagemath")
+    pyproject_metadata = merge_setup_toml_metadata(
+        {}, get_all_toml_info(pyproject_toml)
+    )
+    requirements = extract_requirements(pyproject_metadata, graystull_config, {})
+    all_requirements = (
+        requirements.get("build", [])
+        + requirements.get("host", [])
+        + requirements.get("run", [])
+    )
+
+    # Specify concrete package for some virtual packages
+    all_requirements.remove("{{ blas }}")
+    all_requirements.append("blas=2.*=openblas")
+    all_requirements.remove("{{ compiler('c') }}")
+    all_requirements.append("c-compiler")
+    all_requirements.remove("{{ compiler('cxx') }}")
+    all_requirements.append("cxx-compiler")
+
+    # Correct pypi name for some packages
+    python_requirements = pyproject_metadata.get("install_requires", [])
+    # Specify concrete packages for some packages not yet in grayskull
+    # TODO: It seems to be a bug that these external.dependencies are added to install_requires by grayskull
```

Comment:
Could we reference an upstream issue if we believe that's a bug?

Reply:
I forgot why I thought this would be a bug. I've removed the comment now.